spec/jobs/get_usps_proofing_results_job_spec.rb

```diff
@@ -584,433 +513,568 @@
+              expect(pending_enrollment.enrollment_established_at).to eq(Time.zone.now - 3.days)
+              expect(pending_enrollment.status_updated_at).to eq(Time.zone.now - 1.day)
+              expect(pending_enrollment.status_check_attempted_at).to eq(Time.zone.now)
+              expect(pending_enrollment.status_check_completed_at).to eq(Time.zone.now)
+            end
+
+            expect(pending_enrollment.profile.active).to eq(false)
+            expect(pending_enrollment.pending?).to be_truthy
+
+            expect(job_analytics).not_to have_logged_event(
+              'GetUspsProofingResultsJob: Enrollment status updated',
+            )
+          end
         end
-      end
 
-      context 'when a timeout error occurs' do
-        before(:each) do
-          stub_request_proofing_results_with_timeout_error
+        context 'when a timeout error occurs' do
+          before(:each) do
+            stub_request_proofing_results_with_timeout_error
+          end
+
+          it_behaves_like(
+            'enrollment_encountering_an_error_that_has_a_nil_response',
+            error_type: Faraday::TimeoutError,
+          )
         end
 
-        it_behaves_like(
-          'enrollment_encountering_an_error_that_has_a_nil_response',
-          error_type: Faraday::TimeoutError,
-        )
+        context 'when a nil status error occurs' do
+          before(:each) do
+            stub_request_proofing_results_with_nil_status_error
+          end
+
+          it_behaves_like(
+            'enrollment_encountering_an_error_that_has_a_nil_response',
+            error_type: Faraday::NilStatusError,
+          )
+        end
       end
 
-      context 'when a nil status error occurs' do
-        before(:each) do
-          stub_request_proofing_results_with_nil_status_error
+      describe 'DAV enabled' do
+        let(:capture_secondary_id_enabled) { true }
+        let(:pending_enrollment) do
+          create(
+            :in_person_enrollment, :pending,
+            capture_secondary_id_enabled: capture_secondary_id_enabled
+          )
+        end
+        before do
+          allow(InPersonEnrollment).to receive(:needs_usps_status_check).
+            and_return([pending_enrollment])
+          allow(IdentityConfig.store).to receive(:in_person_proofing_enabled).and_return(true)
         end
 
-        it_behaves_like(
-          'enrollment_encountering_an_error_that_has_a_nil_response',
-          error_type: Faraday::NilStatusError,
-        )
+        context 'when an enrollment passes proofing with an secondary ID and DAV enabled' do
+          before do
+            stub_request_passed_proofing_secondary_id_type_results
+          end
+
+          it_behaves_like(
+            'enrollment_with_a_status_update',
+            passed: false,
+            status: 'failed',
+            response_json: UspsInPersonProofing::Mock::Fixtures.
+              request_passed_proofing_secondary_id_type_results_response,
+          )
```

Comment:
The update to fix the secondary ID check looks good, but I think the test here should be sensitive enough to capture the difference in behavior. We can do that by stubbing the secondary ID with a type that would normally be supported.

Reply:
i updated the json file to use one of the `SUPPORTED_ID` types as the secondary id type